NBitcoin/Protocol/NodesGroup.cs

```diff
@@ -29,43 +29,51 @@ public class WellKnownGroupSelectors
 		}
 
 
-		static Func<IPEndPoint, byte[]> _GroupByIp;
-		public static Func<IPEndPoint, byte[]> ByIp
+		static Func<EndPoint, byte[]> _GroupByIp;
+		public static Func<EndPoint, byte[]> ByIp
 		{
 			get
 			{
-				return _GroupByIp = _GroupByIp ?? new Func<IPEndPoint, byte[]>((ip) =>
+				return _GroupByIp = _GroupByIp ?? new Func<EndPoint, byte[]>((endpoint) =>
 				{
-					return ip.Address.GetAddressBytes();
+					if (endpoint is IPEndPoint ipEndPoint)
+						return ipEndPoint.Address.GetAddressBytes();
+
+					if (endpoint is DnsEndPoint dnsEndPoint)
+					{
+						if (dnsEndPoint.IsTor()) // All Tor endpoints are grouped together
+							return Encoding.UTF8.GetBytes("onion address");
+
+						if (dnsEndPoint.IsI2P()) // All Tor endpoints are grouped together

```

Comment:
Should this comment be:
```suggestion
						if (dnsEndPoint.IsI2P()) // All I2P endpoints are grouped together
```

Reply:
done